rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -291,18 +292,53 @@ fn wireless_config_to_dbus<'a>(
     config: &'a WirelessConfig,
     mac_address: &MacAddress,
 ) -> NestedHash<'a> {
-    let wireless: HashMap<&str, zvariant::Value> = HashMap::from([
+    let mut wireless: HashMap<&str, zvariant::Value> = HashMap::from([
         ("mode", Value::new(config.mode.to_string())),
         ("ssid", Value::new(config.ssid.to_vec())),
         ("assigned-mac-address", Value::new(mac_address.to_string())),
     ]);
 
+    if let Some(band) = &config.band {
+        wireless.insert("band", band.to_string().into());
+        if let Some(channel) = config.channel {
+            wireless.insert("channel", channel.into());
+        }
+    }
+    if let Some(bssid) = &config.bssid {
+        wireless.insert("bssid", bssid.as_bytes().into());
+    }
+
     let mut security: HashMap<&str, zvariant::Value> =
         HashMap::from([("key-mgmt", config.security.to_string().into())]);
 
     if let Some(password) = &config.password {
         security.insert("psk", password.to_string().into());
     }
+    if let Some(wep_security) = &config.wep_security {
+        security.insert(
+            "wep-key-type",
+            (wep_security.wep_key_type.clone() as u32).into(),
+        );
+        security.insert("auth-alg", wep_security.auth_alg.to_string().into());
+        for (i, wep_key) in wep_security.keys.clone().into_iter().enumerate() {
+            security.insert(
+                // lifetimes are fun
```

Comment:
perhaps we should use a `OwnedNestedHash` or just use a `String` as key.

```suggestion
                // FIXME: lifetimes are fun
```